posix/clone

```diff
@@ -1,13 +1,14 @@
 #!/bin/sh
 
-mkdir -p ${DRONE_WORKSPACE}
+sudo mkdir -p ${DRONE_WORKSPACE}
+[ -n "${CI}" ] && sudo chown drone:drone ${DRONE_WORKSPACE}
```

Comment:
I'm thinking we should modify like this, otherwise running unit tests locally on osx would probably block and prompt a password

```
if [[ -n "${CI}" ]]; then
  sudo mkdir -p ${DRONE_WORKSPACE}
  sudo chown drone:drone ${DRONE_WORKSPACE}
else
  mkdir -p ${DRONE_WORKSPACE}
fi
```

Reply:
Done in dd7a4957087edf8dedc5f79fbc1c87957532f1ea.